sbi/analysis/plot.py

```diff
@@ -554,42 +554,56 @@ def handle_nan_infs(samples: List[np.ndarray]) -> List[np.ndarray]:
     return samples
 
 
+def convert_to_list_of_numpy(
+    arr: Union[List[np.ndarray], List[torch.Tensor], np.ndarray, torch.Tensor],
+) -> List[np.ndarray]:
+    """Converts a list of torch.Tensor to a list of np.ndarray."""
+    if not isinstance(arr, list):
+        arr = ensure_numpy(arr)
+        return [arr]
+    return [ensure_numpy(a) for a in arr]
+
+
+def infer_limits(
+    samples: List[np.ndarray], dim: int, points: Optional[List[np.ndarray]] = None
+) -> List:
+    """Infer limits for the plot."""
+    limits = []
+    for d in range(dim):
```

Comment:
do we not need some epsilon added to the limits? Or does this happen somewhere else?

Reply:
good point, as it is now, limits will be right on the edge. I will add a relative epsilon to avoid this.